hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3923,6 +3944,148 @@ DirectoryListing getListing(String src, byte[] startAfter,
     return dl;
   }
 
+  public byte[] getSrcPathsHash(String[] srcs) {
+    MessageDigest md;
+    try {
+      md = MessageDigest.getInstance("MD5");
+    } catch (NoSuchAlgorithmException e) {
+      throw new RuntimeException(e);
+    }
+
+    for (String src: srcs) {
+      md.update(src.getBytes(Charsets.UTF_8));
+    }
+    return md.digest();
+  }
+
+  BatchedDirectoryListing getBatchedListing(String[] srcs, byte[] startAfter,
```

Comment:
This method is longer than I can possibly understand.

Reply:
I found a little hard to refactoring this given that several variables are used throughout the method, such as `lastListing`. Would it be helpful if we add more comments in the method body, explaining what each section is doing?

Reply:
ok. i'm just being lazy.